addons/website/tests/test_website_form_editor.py

```diff
@@ -81,3 +83,25 @@ def test_website_form_html_escaping(self):
             mail = self.env['mail.mail'].search([], order='id desc', limit=1)
             self.assertNotIn('<b>', mail.body_html, "HTML should be escaped in website form")
             self.assertIn('&lt;b&gt;', mail.body_html, "HTML should be escaped in website form (2)")
+
+    def test_website_form_commit_when_creating(self):
+        WebsiteFormController = WebsiteForm()
+        original_insert_record = WebsiteFormController.insert_record
+        def dummy_insert_record(*args, **kwargs):
+            res = original_insert_record(*args, **kwargs)
+            # delete website_form savepoint by rollbacking to test savepoint
+            self.env.cr.execute('ROLLBACK TO SAVEPOINT test_%d' % self._savepoint_id)
+            return res
+        WebsiteFormController.insert_record = dummy_insert_record
+        with MockRequest(self.env):
+            request.params = {
+                'model_name': 'res.partner',
+                'name': 'test partner',
+            }
+            with self.assertLogs(level='ERROR'):
+                response = WebsiteFormController.website_form(
+                    **request.params,
+                )
+            self.assertEqual(response.status_code, 200)
+            self.assertTrue(response.data.startswith(b'{"id":'))
```

Comment:
Your test seems to fail despite your fix while testing in local :thinking:

Reply:
I needed to add this that is set by the editor when testing manually:

```py
self.env.ref('base.model_res_partner').website_form_access = True
self.env['ir.model.fields'].formbuilder_whitelist('res.partner', ['name'])
```